device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -457,18 +457,6 @@ public boolean isEmpty()
         return this.transport.isEmpty();
     }
 
-    /**
-     * Registers a callback with the configured transport to be executed whenever the connection to the device is lost or established.
-     *
-     * @param callback the callback to be called.
-     * @param callbackContext a context to be passed to the callback. Can be
-     * {@code null} if no callback is provided.
-     */
-    public void registerConnectionStateCallback(IotHubConnectionStateCallback callback, Object callbackContext)
```

Comment:
I removed this callback from our device/module clients a while back, but forget to remove a few internal methods tied to it that are no longer needed